src/main/java/com/spotify/docker/client/DefaultDockerClient.java

```diff
@@ -2661,9 +2670,24 @@ public RegistryAuth registryAuth() {
      *
      * @param registryAuth RegistryAuth object
      * @return Builder
+     *
+     * @deprecated in favor of registryAuthSupplier
      */
+    @Deprecated
     public Builder registryAuth(final RegistryAuth registryAuth) {
+      if (this.registryAuthSupplier != null) {
+        throw new IllegalStateException(ERROR_MESSAGE);
+      }
       this.registryAuth = registryAuth;
+      this.registryAuthSupplier = new NoOpRegistryAuthSupplier(registryAuth);
```

Comment:
This line makes calling this method twice throw which might not be obvious to caller if it doesn't know about the implementation here.